src/components/responsive-flyout/ResponsiveFlyout.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import { Flyout, Overlay } from '../flyout';
+import ScrollWrapper from '../scroll-wrapper';
+import type { FlyoutProps } from '../flyout/Flyout';
+import OverlayHeader from './OverlayHeader';
+
+import './styles/ResponsiveFlyout.scss';
+
+type Props = FlyoutProps & {
+    /** Components to render in the overlay */
+    children: ?React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Disable responsiveness behavior for higher-level components that want control */
+    enabled?: boolean,
+    /** Custom button to trigger for opening/closing the flyout */
+    flyoutButton: React.Element<any>,
+    /** Optional function to get the flyoutRef in parent components */
+    flyoutRefFn?: any => any,
+    /** Header for responsive layout */
+    header?: React.Element<any>,
+    /** Overlay classname */
+    overlayClassName?: string,
+    /** Optional function to get the scrollRef in parent components */
+    scrollRefFn?: any => any,
+};
+
+class ResponsiveFlyout extends React.Component<Props> {
```

Comment:
Oh no, a class, haha! If it is because you wanted to use a ref, there is, of course, `useRef`. (You probably know that though.) About refs in general, do we want to reserve refs as last resort (c.f. https://blog.logrocket.com/why-you-should-use-refs-sparingly-in-production/). If so, would it be possible to do this without a ref---and, either way, with a function instead of a <shudder> class? Young guns like me (xp-wise) ain't got no class.

Reply:
That's a good concern to have. I agree its a last resort. I was kind of leaning toward using ResponsiveOverlay at the time of this PR. Might be a good idea to do so